__tests__/InvalidMutations.ts

```diff
@@ -0,0 +1,83 @@
+import { Reactor, State, Bank, App, Triggers, Args, Timer, OutPort, InPort, TimeUnit, TimeValue, Origin, Log, LogLevel, Action } from '../src/core/internal';
+
+class Starter extends Reactor {
+    public out = new OutPort<number>(this);
+
+    constructor(parent: Reactor|null) {
+        super(parent);
+        this.addReaction(
+            new Triggers(this.startup),
+            new Args(this.writable(this.out)),
+            function(this, __out) {
+                __out.set(4);
+
+            }
+        );
+    }
+
+}
+
+class R1 extends Reactor {
+    public in1 = new InPort<number>(this);
+    public in2 = new InPort<number>(this);
+    public out1 = new OutPort<number>(this);
+    public out2 = new OutPort<number>(this);
+
+    constructor(parent: Reactor|null) {
+        super(parent);
+        this.addReaction(
+            new Triggers(this.in1),
+            new Args(this.in1, this.writable(this.out1)),
+            function(this, __in, __out) {
+                __out.set(4)
+            }
+        )
+        this.addReaction(
+            new Triggers(this.in2),
+            new Args(this.in2, this.writable(this.out2)),
+            function(this, __in, __out) {
+                __out.set(4)
+            }
+        )
+
+        this.addMutation(
+            new Triggers(this.in1),
+            new Args(this.in1, this.out2),
+            function(this, __in1, __out2) {
+                test('expect error on mutation creating race condition', () => {
+                    expect(() => {
+                        this.connect(__in1, __out2)
+                    }).toThrowError("New connection introduces direct feed through.")
+                })
+                test('expect error to be thrown on mutation creating loop', () => { 
+                    expect(() => {
+                        this.connect(__out2, __in1)
+                    }).toThrowError("New connection introduces direct feed through and cycle.")
+                })
+                let R2 = new R1(this.getReactor())
+                test('expect error on spawning and creating loop within a reactor', () => {
```

Comment:
This test looks wrong. For the first connect statement to be legal, R1 has to be contained in the reactor that spawned it. For the second connect statement, R1 has to be upstream, which it can't be.

Reply:
Tagging @Wonseo-C

Reply:
https://github.com/lf-lang/reactor-ts/pull/98/commits/58705b73556cde9e1538f951848f4a238262ab91
I made several modifications to the ```canConnect``` logic to fix the issues:
1) I updated the condition that checks for a 'destination port already occupied' error to include all cases, not just when ```this._runtime.isRunning() == false.```
2) I added a new condition to check for connections outside of the container.
3) I made some changes to the direct feed-through condition.


https://github.com/lf-lang/reactor-ts/pull/98/commits/638fd866a569b2da05ab78312979ab95c9650d12
I made an effort to fix the test case by removing the 'reaction2' and allowing for direct feed-through without the 'destination port is already occupied' condition. This was necessary because the previous version of the test case was causing issues with occupied destination ports.